### PR TITLE
Add ado signin

### PR DESCRIPTION
**Pull request by mingwli**
After #75

### CODE REVIEW

**Comment by 928PJY** [comment]
>         sinon = createSandbox();

Memory leak when recreate?
do you need to re-create all of this variable? or just set `environmentController` to return a different docsType?

---
Refers to: test/unitTests/credential/credentialController.test.ts:54 in 167cdf1. [](commit_id = 167cdf1e6dfffb8d9122d15b69ed3f0fbc750b86, deletion_comment = False)

**Comment by mingwli** [comment]
> > ```
> >     sinon = createSandbox();
> > ```
> 
> Memory leak when recreate?
> do you need to re-create all of this variable? or just set `environmentController` to return a different docsType?
> 
> Refers to: test/unitTests/credential/credentialController.test.ts:54 in [167cdf1](https://github.com/docascode/vscode-docs-build/commit/167cdf1e6dfffb8d9122d15b69ed3f0fbc750b86). [](commit_id = [167cdf1](https://github.com/docascode/vscode-docs-build/commit/167cdf1e6dfffb8d9122d15b69ed3f0fbc750b86), deletion_comment = False)

The `CredentialController` and `keyChain` is also needed.
using `sinon.reset()`.

File: test/unitTests/credential/credentialController.test.ts [hunk 1]
```diff
@@ -233,6 +239,109 @@ describe('CredentialController', () => {
         });
     });
 
+
+
+    describe(`User Sign-in With Azure-DevOps`, () => {
+        before(() => {
+            environmentController = getFakeEnvironmentController('Azure DevOps');
+            keyChain = new KeyChain(environmentController);keyChain = new KeyChain(environmentController);
```
Comment (928PJY) [inline test/unitTests/credential/credentialController.test.ts]: >keyChain = new KeyChain(environmentController); [](start = 59, length = 47)

remove?

**Review by 928PJY — APPROVED** [review]
:shipit:

**Pull request merged**